module/VuFindSearch/src/VuFindSearch/Backend/LibGuides/Connector.php

```diff
@@ -187,15 +196,35 @@ protected function process($data)
 
         $items = [];
 
-        // Extract titles and URLs from response:
-        $regex = '/<a href="([^"]*)"[^>]*>([^<]*)</';
-        $count = preg_match_all($regex, $data, $matches);
+        $itemRegex = '/<li>(.*?)<\/li>/';
+        $linkRegex = '/<a href="([^"]*)"[^>]*>([^<]*)</';
+        $descriptionRegex = '/<div class="s-lg-(?:widget|guide)-list-description">(.*?)<\/div>/';
```

Comment:
The class name varies (widget vs. guide) depending whether you are loading research guides or databases.  So this regex works for both.  This seemed simpler than introducing more state and conditional logic.